### PR TITLE
major python dce improvements

### PR DESCRIPTION
**Pull request by frabbit**

### CODE REVIEW

File: dce.ml [hunk 1]
```diff
@@ -356,6 +362,26 @@ and expr dce e =
...
+	| TBinop(OpAdd,e1,e2) when (match follow e.etype with TDynamic _ -> true | _ -> false) ->
+		check_feature dce "add_dynamic";
+		expr dce e1;
+		expr dce e2;
+	| TBinop(OpAssign,({eexpr = TField(_,(FDynamic _ as fa) )} as e1),e2) ->
+		check_dynamic_write dce fa;
+		expr dce e1;
+		expr dce e2;
+	| TBinop(OpAssign,({eexpr = TField(_,(FAnon cf as fa) )} as e1),e2) when Meta.has Meta.Optional cf.cf_meta ->
+		check_dynamic_write dce fa;
```
Comment (Simn) [inline dce.ml]: Could we distinguish FAnon from FDynamic here? Anon access is not the same as dynamic in the general case.

**Comment by frabbit** [comment]
Hello World has now 226 lines, so there is still room for improvements

File: dce.ml (hunk 1 above)
Comment (frabbit) [inline dce.ml]: you mean something like "anon_write", "anon_read"

Comment (Simn) [inline dce.ml]: Yes, or something involving "optional" if you want to highlight that.

Comment (frabbit) [inline dce.ml]: opt_field_write, opt_field_read?

**Comment by nadako** [comment]
guess we wanna squash those commits

**Comment by Simn** [comment]
Indeed!

**Comment by nadako** [comment]
Okay, done. Does this look good to you?

**Pull request merged**